nav2_util/include/nav2_util/twist_publisher.hpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2023 Ryan Friedman
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef NAV2_UTIL__TWIST_PUBLISHER_HPP_
+#define NAV2_UTIL__TWIST_PUBLISHER_HPP_
+
+
+#include <memory>
+#include <string>
+
+#include "geometry_msgs/msg/twist.hpp"
+#include "geometry_msgs/msg/twist_stamped.hpp"
+#include "rclcpp/parameter_service.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/qos.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
+
+#include "lifecycle_node.hpp"
+#include "node_utils.hpp"
+
+namespace nav2_util
+{
+
+/**
+ * @class nav2_util::TwistPublisher
+ * @brief A simple wrapper on a Twist publisher that provides either Twist or TwistStamped
+ *
+ * The default is to publish Twist to preserve backwards compatibility, but it can be overridden
+ * using the "enable_stamped_cmd_vel" parameter to publish TwistStamped.
+ *
+ */
+
+class TwistPublisher
+{
+public:
+  /**
+  * @brief A constructor
+  * @param nh The node
+  * @param topic publisher topic name
+  * @param qos publisher quality of service
+  */
+  explicit TwistPublisher(
+    nav2_util::LifecycleNode::SharedPtr node,
+    const std::string & topic,
+    const rclcpp::QoS & qos
+  )
```

Comment:
`)` on line above, this looks like a strange auto-reformatting choice

Reply:
Interesting. ament_uncrustify accepts code in either configuration as valid. Sorry, I'm used to clang-tidy which always results in the same output regardless of what the input code is.

Fixed!

Reply:
e65fed5e66dac9f0cc264766c078b8b31332f1cd